---
title: "RNA-seq analysis of physalis samples"
output: rmarkdown::github_document
---

```{r, echo = FALSE}
knitr::opts_chunk$set(
  fig.path = "figures/rnaseq-"
)
```

Download datasets

```{bash, message=FALSE, warning=FALSE}
mkdir input
wget -O input/elicitors.salmon.merged.gene_counts.tsv https://zenodo.org/records/13902262/files/elicitors.salmon.merged.gene_counts.tsv?download=1
wget -O input/elicitors.salmon.merged.gene_tpm.tsv https://zenodo.org/records/13902262/files/elicitors.salmon.merged.gene_tpm.tsv?download=1
wget -O input/tissues.salmon.merged.gene_counts.tsv https://zenodo.org/records/13902262/files/tissues.salmon.merged.gene_counts.tsv?download=1
wget -O input/tissues.salmon.merged.gene_tpm.tsv https://zenodo.org/records/13902262/files/tissues.salmon.merged.gene_tpm.tsv?download=1
```



```{r}
library(tidyverse)
```

```{r}
sessionInfo()
```


# Figure 1A and related suplementary figures (2A, 2B and 2E)

## Supplementary figure 2A and 2B

Outliers:

- "X9_3_S28" fruit 3
- "X4_1_S3" young leaf1

```{r, message=FALSE}

tissues_counts <- read_tsv("input/tissues.salmon.merged.gene_counts.tsv") %>%
  dplyr::select(!c('gene_id')) %>%
  mutate(gene_name = gsub("gene:", "", gene_name)) %>% 
  column_to_rownames("gene_name") %>%
  as.matrix() %>%
  round()

tissues_counts_no_outliers <- read_tsv("input/tissues.salmon.merged.gene_counts.tsv") %>%
  dplyr::select(!c('gene_id', 'X9_3_S28', 'X4_1_S3')) %>%
  mutate(gene_name = gsub("gene:", "", gene_name)) %>% 
  column_to_rownames("gene_name") %>%
  as.matrix() %>%
  round()

```

Pre-filtering low count genes

```{r, message=FALSE} 
tissues_counts <- tissues_counts[rowSums(tissues_counts) > 10,]
tissues_counts_no_outliers <- tissues_counts_no_outliers[rowSums(tissues_counts_no_outliers) > 10,]
```


Metadata

```{r, message=FALSE}
tissues_metadata <- data.frame(SampleName = colnames(tissues_counts)) %>%
  separate(SampleName, c("Condition", "Replicate"), remove = F) %>%
  mutate(Condition = gsub("X10", "Calyx", Condition)) %>%
  mutate(Condition = gsub("X1", "Root", Condition)) %>%
  mutate(Condition = gsub("X2", "Shoot", Condition)) %>%
  mutate(Condition = gsub("X4", "Young_leaf", Condition)) %>%
  mutate(Condition = gsub("X5", "Large_leaf", Condition)) %>%
  mutate(Condition = gsub("X6", "Petiole", Condition)) %>%
  mutate(Condition = gsub("X7", "Stem", Condition)) %>%
  mutate(Condition = gsub("X8", "Flower", Condition)) %>%
  mutate(Condition = gsub("X9", "Fruit", Condition))  %>% 
  column_to_rownames("SampleName")

tissues_metadata_no_outliers <- data.frame(SampleName = colnames(tissues_counts_no_outliers)) %>%
  separate(SampleName, c("Condition", "Replicate"), remove = F) %>%
  mutate(Condition = gsub("X10", "Calyx", Condition)) %>%
  mutate(Condition = gsub("X1", "Root", Condition)) %>%
  mutate(Condition = gsub("X2", "Shoot", Condition)) %>%
  mutate(Condition = gsub("X4", "Young_leaf", Condition)) %>%
  mutate(Condition = gsub("X5", "Large_leaf", Condition)) %>%
  mutate(Condition = gsub("X6", "Petiole", Condition)) %>%
  mutate(Condition = gsub("X7", "Stem", Condition)) %>%
  mutate(Condition = gsub("X8", "Flower", Condition)) %>%
  mutate(Condition = gsub("X9", "Fruit", Condition))  %>% 
  column_to_rownames("SampleName")
  
```

Checking that input data for DESeq2 is correct. It is critical that the columns of the count matrix and the rows of the column data (information about samples) are in the same order

```{r, message=FALSE}
all(rownames(tissues_metadata) %in% colnames(tissues_counts))
```

```{r, message=FALSE}
all(rownames(tissues_metadata_no_outliers) %in% colnames(tissues_counts_no_outliers))

```


Input DESeqDataSet

```{r, message=FALSE}
dds_tissues_counts <- DESeq2::DESeqDataSetFromMatrix(countData = tissues_counts,
                                   colData = tissues_metadata,
                                   design = ~ Condition)
dds_tissues_counts
```

```{r, message=FALSE}
dds_tissues_counts_no_outliers <- DESeq2::DESeqDataSetFromMatrix(countData = tissues_counts_no_outliers,
                                   colData = tissues_metadata_no_outliers,
                                   design = ~ Condition)
dds_tissues_counts_no_outliers
```
Count normalisation

```{r, message=FALSE}
vst_tissue <- DESeq2::vst(dds_tissues_counts)
vst_tissue_no_outliers <- DESeq2::vst(dds_tissues_counts_no_outliers)
```

### Plot PCA (2A)

Tissue with outliers
```{r, message=FALSE}
DESeq2::plotPCA(vst_tissue, intgroup = c("Condition", "Replicate"), ntop = 500)
```


```{r SupFig2A, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}

DESeq2::plotPCA(vst_tissue, intgroup = c("Condition", "Replicate"), returnData = T, ntop = 500)  %>%
  ggplot(aes(x = PC1, y=PC2, fill = Condition, label = as.factor(Replicate))) +
  geom_point(size = 4, pch = 21, 
             alpha = 0.7
             ) +
  geom_text() +
  scale_fill_manual(values = MoMAColors::moma.colors('Rattner', n = 9, return_hex = TRUE)) +
  theme_classic() +
  theme(text=element_text(family = "Helvetica", size =14)) +
  labs(title = "",
       x = "PC1 [39%]",
       y = "PC2 [18%]",
       fill = "Tissue")
```

```{r, message=FALSE}
 ggsave('figures/tissue_pca.pdf', 
        height = 4,
        width = 6,
        bg = 'white')

 ggsave('figures/tissue_pca.svg', 
        height = 4,
        width = 6,
        bg = 'white')
```

### Plot PCA (2B)

Tissue with outliers
```{r, message=FALSE}
DESeq2::plotPCA(vst_tissue_no_outliers, intgroup = c("Condition", "Replicate"), ntop = 500)
```

Tissue No outliers
```{r SupFig2B, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}

DESeq2::plotPCA(vst_tissue_no_outliers, intgroup = c("Condition", "Replicate"), returnData = T, ntop = 500)  %>%
  ggplot(aes(x = PC1, y=PC2, fill = Condition, label = as.factor(Replicate))) +
  geom_point(size = 4, pch = 21, 
             alpha = 0.7
             ) +
  geom_text() +
  scale_fill_manual(values = MoMAColors::moma.colors('Rattner', n = 9, return_hex = TRUE)) +
  theme_classic() +
  theme(text=element_text(family = "Helvetica", size =14)) +
  labs(title = "",
       x = "PC1 [38%]",
       y = "PC2 [22%]",
       fill = "Tissue")
```

```{r, message=FALSE}
 ggsave('figures/tissue_pca_no_outliers.pdf', 
        height = 4,
        width = 6,
        bg = 'white')

 ggsave('figures/tissue_pca_no_outliers.svg', 
        height = 4,
        width = 6,
        bg = 'white')
```


## Supplementary figure 2C

```{r, message=FALSE}
elicitors_counts <- 
  read_tsv("input/elicitors.salmon.merged.gene_counts.tsv") %>%
  dplyr::select(!('gene_id')) %>%
  mutate(gene_name = gsub("gene:", "", gene_name)) %>% 
  column_to_rownames("gene_name") %>%
  rename_with(~ gsub("JA", "MeJA", .x, fixed = TRUE)) %>%
  as.matrix() %>%
  round()
```

Pre-filtering low count genes

```{r, message=FALSE} 
elicitors_counts <- elicitors_counts[rowSums(elicitors_counts) > 10,]
```

Metadata

```{r, message=FALSE}
elicitors_metadata <- data.frame(SampleName = colnames(elicitors_counts)) %>%
  separate(SampleName, c("Tissue", "Condition", "Replicate"), remove = F) %>%
  column_to_rownames("SampleName")

```

Checking that input data for DESeq2 is correct

```{r, message=FALSE}
all(rownames(elicitors_metadata) %in% colnames(elicitors_counts))
```

```{r, message=FALSE}
dds_elicitors <- DESeq2::DESeqDataSetFromMatrix(countData = elicitors_counts,
                                   colData = elicitors_metadata,
                                   design = ~ Tissue + Condition + Tissue:Condition) 

dds_elicitors
```

Normalisation

```{r, message=FALSE}
vst_elicitors <- DESeq2::vst(dds_elicitors)
```

#### Plot PCA (2C)

```{r, message=FALSE}
DESeq2::plotPCA(vst_elicitors, intgroup = c("Tissue", "Condition"), ntop = 500)
```


```{r SupFig2C, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}
DESeq2::plotPCA(vst_elicitors, intgroup = c("Tissue", "Condition"), returnData = T, ntop = 500)  %>%
  ggplot(aes(x = PC1, y=PC2, fill = Condition)) +
  geom_point(size = 3, pch = 21, alpha = 0.7) +
  facet_grid(~Tissue) +
  scale_fill_manual(values = MoMAColors::moma.colors('Rattner', n = 4, return_hex = TRUE)) +
  theme_classic() +
  theme(text = element_text(family = "Helvetica", size =14) ) +
  labs(title = "",
       x = "PC1 [79%]",
       y = "PC2 [13%]",
       fill = "Treatment")
```

```{r, message=FALSE}
 ggsave('figures/elicitors_pca.pdf', 
        height = 3,
        width = 5,
        bg = 'white')

 ggsave('figures/elicitors_pca.svg', 
        height = 3,
        width = 5,
        bg = 'white')
```


## Figure 1B

```{r, message=FALSE}
tissues_tpm <- read_tsv("input/tissues.salmon.merged.gene_tpm.tsv") %>%
  dplyr::select(!('gene_id')) %>%
  mutate(gene_name = gsub("gene:", "", gene_name))
```

Compute Z-score, remove outliers, tidy data and average tpm per tissue

```{r, message=FALSE}

outliers = c('X9_3_S28' , 'X4_1_S3')

td_tissues_tpm <- tissues_tpm %>%
  dplyr::select(!all_of(outliers)) %>%
  pivot_longer(cols = !gene_name, names_to = "Sample", values_to = "TPM") %>%
  separate(Sample, c("Tissue", "Replicate", "Run"), sep="_", extra="drop", remove=F) %>%
  dplyr::select(!c("Sample", "Run","Replicate")) %>%
  group_by(gene_name, Tissue) %>%
  summarise_all(mean) %>%
  ungroup()
  
```


Filter out genes with TPM < 10 in total

```{r, message=FALSE}
expressed_genes_tissues <- td_tissues_tpm %>% 
  dplyr::select(gene_name, TPM) %>%
  group_by(gene_name) %>%
  summarise_all(sum) %>%
  filter(TPM >= 10)

```

zscore of expressed genes


```{r, message=FALSE}
z_score_expressed_genes_tissue <- td_tissues_tpm %>%
  filter(gene_name %in% expressed_genes_tissues$gene_name) %>%
  group_by(gene_name) %>% 
  mutate(z_score = (TPM - mean(TPM)) / sd(TPM)) %>%
  ungroup %>%
  dplyr::select(gene_name, Tissue, z_score)

```


Appending z score to the main dataframe

```{r, message=FALSE}

td_zs_tissues <- left_join(td_tissues_tpm, z_score_expressed_genes_tissue)

```

Renaming and reordering tissue

```{r, message=FALSE}
tissue_code <- c("X1","X2","X4","X5","X6","X7","X8","X9","X10")
tissue_name <- c("2w_roots", "2w_shoot", "young_leaf", "large_leaf","petiole","stem","flower","fruit", "calyx")

tissue_df <- data_frame(tissue_code, tissue_name) %>% 
  magrittr::set_colnames(c("Tissue","Tissue_Name"))

td_zs_tissues <- left_join(td_zs_tissues, tissue_df)

tissue_order <- c("2w_roots",
                  "petiole",
                  "fruit",
                  "flower",
                  "calyx",
                  "stem",
                  "large_leaf",
                  "2w_shoot",
                  "young_leaf")
```

Defining cluster genes

```{r, message=FALSE}
# RENAME THE GENES

cluster_genes_id = c('Phygri01g018760','Phygri01g018780','Phygri01g018790','Phygri01g018800','Phygri01g018810','Phygri01g018820','Phygri01g018840','Phygri01g018850','Phygri01g018880','Phygri01g018890','Phygri01g018900','Phygri01g018920','Phygri01g018930','Phygri01g018940','Phygri01g018950','Phygri01g018960','Phygri01g018970','Phygri01g018980','Phygri01g018990', 'Phygri01g019010','Phygri01g019020','Phygri01g019040')

cluster_genes_name = c('CYP87G1b','24ISOb', '2ODD1','2ODD2', '2ODD4', '2ODD5', 'Sulfb', 'CYP749B2b','CYP88C8', '2ODD3', 'CYP88C7b', 'CYP88C10b', 'SDRb', 'ASAT', '2ODD6', 'CYP749B2a', 'CYP88C7a', 'CYP88C10a', 'CYP87G1a', '24ISOa', 'SDRa', 'Sulfa')

cluster_genes_expression = c("R", "R","R","R","R","R","R","R","R","R","R","R","R","R","L", "L","L","L","L","L","L","L")

cluster_genes_df <- data_frame(cluster_genes_id, cluster_genes_name, cluster_genes_expression) %>% 
  magrittr::set_colnames(c("gene_name","Gene_Name", "Expression_Pattern"))

```


#### Plot

```{r Fig1B, fig.height=3, fig.width=9, message=FALSE, warning=FALSE}
td_zs_tissues %>%
  filter(gene_name %in% cluster_genes_id) %>%
  left_join(cluster_genes_df) %>%
  ggplot(aes(y = fct_relevel(Tissue_Name, tissue_order), 
             x = fct_relevel(Gene_Name, cluster_genes_name),
             fill = z_score)) +
  facet_grid(~factor(Expression_Pattern, levels = c("R", "L")), 
             drop = T, 
             scales="free_x", 
             space="free_x") +
  scale_fill_gradient2(low = "#3F7373", 
                      mid = "white",
                      high = "#A64826",
                      midpoint = 0,
                      limits = c(-2,2),
                      oob = scales::squish,
                      name = "z score" ) +
  labs(x = "", 
       y = "") +
  scale_x_discrete(position = "top") +
  theme_classic() +
  theme(
        text = element_text(family = "Helvetica", size = 14),
        axis.text.x = element_text(size = 8, angle = 45, vjust = 0.5, hjust=0),
        strip.text.x = element_blank(),
        panel.border = element_rect(fill = NA,
                                    linewidth = 1,
                                    color = "black"),
        panel.spacing = unit(0.2, "lines")
        ) +
  geom_tile()

```

```{r, message=FALSE}
ggsave('figures/Tissue_heatmap.svg',
       bg = 'white',
       width = 9,
       height = 3)

ggsave('figures/Tissue_heatmap.pdf',
       bg = 'white',
       width = 9,
       height = 3)
```


## Supplementary figure 2E

```{r SupFig2E, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}
td_zs_tissues %>%
  filter(gene_name %in% cluster_genes_id) %>%
  left_join(cluster_genes_df) %>%
  ggplot(aes(y = fct_relevel(Tissue_Name, tissue_order), 
             x = fct_relevel(Gene_Name, cluster_genes_name),
             fill = TPM)) +
  facet_grid(~factor(Expression_Pattern, levels = c("R", "L")), 
             drop = T, 
             scales="free_x", 
             space="free_x") +
  scale_fill_viridis_c(option = 'mako',
                       trans = "log",
                       breaks = scales::trans_breaks("log10", function(x) 10^x)(c(1, 1e6)),
                       oob = scales::squish ) + 
                     
  labs(x = "", 
       y = "") +
  scale_x_discrete(position = "top") + 
  theme_classic() +
  theme(
        text = element_text(family = "Helvetica", size = 12),
        axis.text.x = element_text(size = 10, angle = 45, vjust = 0.5, hjust=0),
        axis.text.y = element_text(size = 12),
        strip.text.x = element_blank(),
        panel.border = element_rect(fill = NA,
                                    linewidth = 1,
                                    color = "black"),
        panel.spacing = unit(0.2, "lines")
        ) +
  geom_tile()
```

```{r, message=FALSE}
ggsave('figures/Tissue_heatmap_tpm.svg',
       bg = 'white',
       width = 6,
       height = 4)

ggsave('figures/Tissue_heatmap_tpm.pdf',
       bg = 'white',
       width = 6,
       height = 4)
```


## Figure 1C


```{r, message=FALSE}
##### SHOOT ####
dds_elicitors_shoot <- DESeq2::DESeqDataSetFromMatrix(countData = elicitors_counts %>%
                                                        as.data.frame() %>%
                                                        dplyr::select(starts_with("Shoot")) %>%
                                                        as.matrix(),
                                   colData = elicitors_metadata %>%
                                     dplyr::filter(Tissue == 'Shoot') %>%
                                     dplyr::select(!Tissue),
                                   design = ~ Condition)

##### ROOT ####
dds_elicitors_root <- DESeq2::DESeqDataSetFromMatrix(countData = elicitors_counts %>%
                                                        as.data.frame() %>%
                                                        dplyr::select(starts_with("Root")) %>%
                                                        as.matrix(),
                                   colData = elicitors_metadata %>%
                                     dplyr::filter(Tissue == 'Root') %>%
                                     dplyr::select(!Tissue),
                                   design = ~ Condition)


```

Set reference 

```{r, message=FALSE}
#### SHOOT ####
dds_elicitors_shoot$Condition <- relevel(dds_elicitors_shoot$Condition, ref = "Mock")

#### ROOT ####
dds_elicitors_root$Condition <- relevel(dds_elicitors_root$Condition, ref = "Mock")

```

Run Differentially Expression Analysis

```{r, message=FALSE}
##### SHOOT #####
dds_elicitors_shoot_DESeq <- DESeq2::DESeq(dds_elicitors_shoot)

##### ROOT #####
dds_elicitors_root_DESeq <- DESeq2::DESeq(dds_elicitors_root)

```

See groups

Shoot
```{r, message=FALSE}
DESeq2::resultsNames(dds_elicitors_shoot_DESeq)
```

Root 
```{r, message=FALSE}
DESeq2::resultsNames(dds_elicitors_root_DESeq)
```


Extract results DESeq

```{r, message=FALSE}
### SHOOT ###

all_l2fcs_shoot <- rbind(
  DESeq2::results(dds_elicitors_shoot_DESeq, c("Condition", "MeJA", "Mock"), tidy = T) %>%
    mutate(Comparison = "Shoot_MeJA"),
  DESeq2::results(dds_elicitors_shoot_DESeq, c("Condition", "Flg22", "Mock"), tidy = T) %>%
    mutate(Comparison = "Shoot_Flg22"),
  DESeq2::results(dds_elicitors_shoot_DESeq, c("Condition", "ABA", "Mock"), tidy = T) %>%
    mutate(Comparison = "Shoot_ABA")
) %>%
  # Adjust to account for large number of comparisons
  mutate(padj_global = p.adjust(pvalue,"fdr") ) %>%
  mutate(gene_name = row)


### ROOT ###

all_l2fcs_root <- rbind(
  DESeq2::results(dds_elicitors_root_DESeq, c("Condition", "MeJA", "Mock"), tidy = T) %>%
    mutate(Comparison = "Root_MeJA"),
  DESeq2::results(dds_elicitors_root_DESeq, c("Condition", "Flg22", "Mock"), tidy = T) %>%
    mutate(Comparison = "Root_Flg22"),
  DESeq2::results(dds_elicitors_root_DESeq, c("Condition", "ABA", "Mock"), tidy = T) %>%
    mutate(Comparison = "Root_ABA")
) %>%
  # Adjust to account for large number of comparisons
  mutate(padj_global = p.adjust(pvalue,"fdr") ) %>%
  mutate(gene_name = row)


### Both Combined

all_l2fcs_sep <- rbind(all_l2fcs_shoot, all_l2fcs_root)

```


### Plot


```{r Fig1C, fig.height=2.5, fig.width=7, message=FALSE, warning=FALSE}
all_l2fcs_sep %>%
  filter(gene_name %in% cluster_genes_id) %>%
  left_join(cluster_genes_df) %>%
  separate("Comparison", c("Tissue", "Treatment"),"_", remove=F) %>% 
  # set DEG
  mutate(DEG = case_when(
    padj_global < 0.01 & log2FoldChange >= 1 ~ paste0("UP"),
    padj_global < 0.01 & log2FoldChange <= -1 ~ paste0("DW"),
    padj_global > 0.01 ~ paste0("NoDiff"),
    log2FoldChange >= -1 & log2FoldChange <= 1 ~ paste0("NoDiff"),
    is.na(padj_global) ~ paste0("NoDiff")
                       ) ) %>%
  mutate(log2FoldChange = case_when(DEG == "NoDiff" ~ NA_real_,
                                    TRUE ~ log2FoldChange) 
         ) %>%
  ggplot(aes(y = Treatment,
             x = fct_relevel(Gene_Name, cluster_genes_name), # setting the order
             fill = log2FoldChange)) +
  facet_grid(cols = vars(factor(Expression_Pattern, levels = c("R", "L"))),
             rows = vars(factor(Tissue, levels = c("Shoot", "Root"))),
             drop = T,
             switch = "y",
             scales="free", 
             space="free") +
  scale_fill_gradientn(colors = scico::scico(9, palette = 'berlin'),
                      na.value = "#161919",
                      limits = c(-4,4),
                      oob = scales::squish,
                      name = "log2(Fold Change)" ) +
  labs(x = "", 
       y = "") +
  scale_x_discrete(position = "top") + 
  theme_classic() +
  theme(
        text = element_text(family = "Helvetica", size = 14),
        axis.text.x = element_text(angle = 45, vjust = 0.5, hjust=0),
        strip.text.x = element_blank(),
        strip.background = element_rect(colour=NA, fill=NA),
        panel.border = element_rect(fill = NA, 
                                    linewidth = 1,
                                    color = "black"),
        panel.spacing = unit(0.2, "lines"),
        ) +
  geom_tile()

```

```{r, message=FALSE}
ggsave('figures/Elicitors_LFC_heatmap.pdf', 
       bg = 'white',
       width = 7,
       height = 2.5)
```


## Supplementary figure 2F

```{r, message=FALSE}
elicitors_tpm <- 
  read_tsv("input/elicitors.salmon.merged.gene_tpm.tsv") %>%
  rename_with(~ gsub("JA", "MeJA", .x, fixed = TRUE)) %>%
  dplyr::select(!('gene_name'))

```
Mean tpm per replicate

```{r, message=FALSE}
td_elicitors_tpm <- elicitors_tpm %>%
  mutate(gene_name = gsub("gene:", "", gene_id)) %>%
  dplyr::select(!gene_id) %>%
  pivot_longer(cols = !gene_name, names_to = "Sample", values_to = "TPM") %>%
  separate(Sample, c("Tissue", "Treatment", "Replicate"), sep="_", extra="drop", remove=F) %>%
  dplyr::select(!c("Sample", "Replicate")) %>%
  group_by(gene_name, Tissue, Treatment) %>%
  summarise_all(mean) %>%
  ungroup()


```

FILTER OUT GENES WITH TPM < 10 IN TOTAL

```{r, message=FALSE}
expressed_genes_elicitors <- td_elicitors_tpm %>% 
  dplyr::select(!c("Tissue", "Treatment")) %>%
  group_by(gene_name) %>%
  summarise_all(sum) %>%
  filter(TPM >= 10)
```

zscore of expressed genes

```{r, message=FALSE}
z_score_expressed_genes_td_elicitors_tpm <- td_elicitors_tpm %>%
  filter(gene_name %in% expressed_genes_elicitors$gene_name) %>%
  group_by(gene_name) %>% 
  mutate(z_score = (TPM - mean(TPM)) / sd(TPM)) %>%
  ungroup %>%
  select(gene_name, Tissue, Treatment, z_score)
```


Appending z score to the main dataframe

```{r, message=FALSE}

zs_td_elicitors_tpm <- left_join(td_elicitors_tpm, z_score_expressed_genes_td_elicitors_tpm) 
```


### Plot

```{r, message=FALSE}
tissue_order_elicitors <- c("Root_ABA",
                  "Root_Flg22", 
                  "Root_MeJA",
                  "Root_Mock",
                  "Shoot_ABA",
                  "Shoot_Flg22",
                  "Shoot_MeJA",
                  "Shoot_Mock")
```


Heatmap tpm elicitors

```{r SupFig2F, fig.height=4, fig.width=6, message=FALSE, warning=FALSE}
zs_td_elicitors_tpm %>%
  filter(gene_name %in% cluster_genes_id) %>%
  left_join(cluster_genes_df) %>%
  ggplot(aes(y = Treatment,
             x = fct_relevel(Gene_Name, cluster_genes_name), 
             fill = TPM)) +
  facet_grid(cols = vars(factor(Expression_Pattern, levels = c("R", "L"))),
             rows = vars(factor(Tissue, levels = c("Shoot", "Root"))),
             drop = T,
             switch = "y",
             scales="free", 
             space="free") +
  scale_fill_viridis_c(option = 'mako',
                       trans = "log",
                       breaks = scales::trans_breaks("log10", function(x) 10^x)(c(1, 1e6)),
                       oob = scales::squish,
                       na.value = "black") +
  labs(x = "", 
       y = "") +
  scale_x_discrete(position = "top") +
  theme_classic() +
  theme(
        text = element_text(family = "Helvetica", size = 14),
        axis.text.x = element_text(size = 10, angle = 45, vjust = 0.5, hjust=0),
        strip.text.x = element_blank(),
        panel.border = element_rect(fill = NA,
                                    linewidth = 1,
                                    color = "black"),
        panel.spacing = unit(0.2, "lines")
        ) +
  geom_tile()

```

```{r, message=FALSE}
ggsave('figures/elicitors_heatmap_tpm.svg',
       bg = 'white',
       width = 6,
       height = 4)

ggsave('figures/elicitors_heatmap_tpm.pdf',
       bg = 'white',
       width = 6,
       height = 4)
```

## Supplementary Figure 2D 


Load Expression as Non-normalised Counts


Tissue

```{r, message=FALSE}
tissues_counts_cemitool <- read_tsv("input/tissues.salmon.merged.gene_counts.tsv") %>%
  dplyr::select(!('gene_id')) %>%
  mutate(Gene_Name= gsub("gene:", "", gene_name)) %>%
  select(!c("gene_name", "X9_3_S28", "X4_1_S3"))
```

Elicitors

```{r, message=FALSE}
elicitor_counts_cemitool <- read_tsv("input/elicitors.salmon.merged.gene_counts.tsv") %>%
  dplyr::select(!('gene_id')) %>%
  mutate(Gene_Name= gsub("gene:", "", gene_name)) %>%
  select(!gene_name)

```

Combined datasets

```{r, message=FALSE}
counts_datasets_physalis <- dplyr::left_join(tissues_counts_cemitool, 
                                    elicitor_counts_cemitool, 
                                    by = "Gene_Name",
                                    keep = TRUE) %>% 
                    mutate(Gene_Name = Gene_Name.x) %>%
                    select(!c("Gene_Name.x", "Gene_Name.y")) %>%
                    column_to_rownames("Gene_Name") %>%
                    as.matrix() %>%
                    round()

```

Normalise Counts
```{r, message=FALSE}
counts_datasets_physalis_norm <- 
  DESeq2::vst(counts_datasets_physalis, blind = TRUE) %>%
                        as.data.frame()
```

Experiment Metadata

```{r, message=FALSE}
tissue_code <- c("X1","X2","X4","X5","X6","X7","X8","X9","X10")
tissue_name <- c("2w_roots", "2w_shoot", "young_leaf", "old_leaf","petiole","stem","flower","fruit", "calyx")
```

```{r, message=FALSE}
Physalis_metadata <- data.frame(SampleName = colnames(counts_datasets_physalis_norm)) %>%
  separate(SampleName, c("Tissue", "Condition"), remove = F) %>%
  mutate(Condition = gsub("22", "", Condition)) %>%
  mutate(Condition = gsub("1|2|3|4|5", "No-Treatment", Condition)) %>%
  mutate(Tissue = gsub("X10", "Calyx", Tissue)) %>%
  mutate(Tissue = gsub("X1", "Root", Tissue)) %>%
  mutate(Tissue = gsub("X2", "Shoot", Tissue)) %>%
  mutate(Tissue = gsub("X4", "Small_Leave", Tissue)) %>%
  mutate(Tissue = gsub("X5", "Large_Leave", Tissue)) %>%
  mutate(Tissue = gsub("X6", "Petiol", Tissue)) %>%
  mutate(Tissue = gsub("X7", "Stem_Leave", Tissue)) %>%
  mutate(Tissue = gsub("X8", "Flower", Tissue)) %>%
  mutate(Tissue = gsub("X9", "Fruit", Tissue)) %>%
  unite("Class", Tissue:Condition, remove = FALSE)
```


Run cemitool
Careful, CEMiTool usually crashes if DESeq2 is loaded.

```{r, message=FALSE}
set.seed(78235)

Physalis_cem <- CEMiTool::cemitool(counts_datasets_physalis_norm,
                                   annot = Physalis_metadata,
                                   network_type = "signed",
                                   filter = TRUE,
                                   filter_pval = 0.1,
                                   apply_vst = FALSE,
                                   cor_method = "pearson",
                                   )
```

Modules summary

```{r, message=FALSE}
# Summarizes modules using mean or eigengene expression.

# If not indicated uses mean instead of eigengene 

mod_eigengenes <- CEMiTool::mod_summary(Physalis_cem, method = 'mean') %>%
  pivot_longer(!modules, names_to = "condition", values_to = "mean_expression") %>%
  separate(condition, c("Tissue", "Treatment"), remove = F) %>%
  mutate(Treatment = gsub("22", "", Treatment)) %>%
  mutate(Treatment = gsub("1|2|3|4|5", "NA", Treatment)) %>%
  mutate(Tissue = gsub("X10", "Calyx", Tissue)) %>%
  mutate(Tissue = gsub("X1", "2wRoot", Tissue)) %>%
  mutate(Tissue = gsub("X2", "2wShoot", Tissue)) %>%
  mutate(Tissue = gsub("X4", "Small.Leave", Tissue)) %>%
  mutate(Tissue = gsub("X5", "Large.Leave", Tissue)) %>%
  mutate(Tissue = gsub("X6", "Petioll", Tissue)) %>%
  mutate(Tissue = gsub("X7", "Stem.Leave", Tissue)) %>%
  mutate(Tissue = gsub("X8", "Flower", Tissue)) %>%
  mutate(Tissue = gsub("X9", "Fruit", Tissue)) %>%
  unite("Class", Tissue:Treatment, remove = FALSE) %>%
  mutate(Class = gsub("_NA", "", Class)) %>%
  dplyr::select(modules, Class, mean_expression) %>% 
  group_by(modules, Class) %>%
  summarise_all(mean)

  
```


It is possible that the modules containing the root or shoot subcluster are named differently than in the manuscript. To find in which module, each subcluster is: 

```{r, message=FALSE}
root_subcluster <-c('Phygri01g018760','Phygri01g018780','Phygri01g018790','Phygri01g018800','Phygri01g018810','Phygri01g018820','Phygri01g018840','Phygri01g018850','Phygri01g018880','Phygri01g018890','Phygri01g018900','Phygri01g018920','Phygri01g018930','Phygri01g018940')

shoot_subcluster <- c('Phygri01g018950','Phygri01g018960','Phygri01g018970','Phygri01g018980','Phygri01g018990','Phygri01g019000','Phygri01g019010','Phygri01g019020','Phygri01g019040')
```

```{r, message=FALSE}
CEMiTool::module_genes(Physalis_cem) %>%
  filter(genes %in% shoot_subcluster)
```

```{r, message=FALSE}
CEMiTool::module_genes(Physalis_cem) %>%
  filter(genes %in% root_subcluster)
```



```{r SupFig2D, fig.height=4, fig.width=7, message=FALSE, warning=FALSE}
modules_clustering <- mod_eigengenes %>% pivot_wider(names_from = Class, values_from = mean_expression) %>% column_to_rownames("modules") %>% as.matrix() %>% dist() %>% hclust()

Class_clustering <- c("Root_ABA", "Root_JA", "Root_Flg", "Root_Mock", "2wRoot", "Fruit", "Shoot_JA", "Shoot_ABA", "Shoot_Flg", "Shoot_Mock", "Flower", "Calyx", "Petioll", "Stem.Leave", "2wShoot", "Large.Leave", "Small.Leave")

mod_eigengenes %>%
  ggplot(aes(x = fct_relevel(Class,
                             Class_clustering),
             y =  fct_relevel(modules,
                            modules_clustering$labels),
             fill = mean_expression)) +
  scale_fill_gradientn(colors = RColorBrewer::brewer.pal(9, "PuBuGn")) +
  theme_classic() +
  labs(x = "",
       y = "") +
  theme(
        text = element_text(family = "Helvetica", size = 14),
        axis.text.x = element_text(angle = 45, vjust = 1, hjust=1),
        panel.border = element_rect(fill = NA, 
                                    linewidth = 1,
                                    color = "black"),) +
  geom_tile()
```

```{r, message=FALSE}
ggsave('figures/Modules_heatmap.pdf', bg = 'white', width = 7, height = 4)
ggsave('figures/Modules_heatmap.svg', bg = 'white', width = 7, height = 4)
```





# Remove input datasets

```{bash}
rm -r input
```

